### PR TITLE
fix(lsp): Add separator between codelens in virttext

### LINKED ISSUE #15048: Multiple LSP code-lenses are concatenated without visual separator
- `nvim --version`: `v0.5.0`
- language server name/version: `gopls 0.7.0`
- Operating system/version: `macOS 11.4`

I simply set `[[autocmd CursorHold,CursorHoldI,InsertLeave <buffer> lua vim.lsp.codelens.refresh()]]` in my Lua config to set up codelens support. If there is configuration to do this already then I couldn't find it, and I apologise. I am not sure if this is a bug. I just went with the ticket template that felt the closest. It's not strictly broken, but it could be nicer, I guess?

### Actual behaviour

The code lens actions are all concatenated into one string and there is no visual separator between these.
`check for UpdatesUpgrade transitive dependenciesUpgrade direct dependencies`
It looks like this:
Screenshot 2021-07-10 at 18 51 10

### Expected behaviour
Multiple items in a codelens list are visually separated
e.g.: `check for Updates | Upgrade transitive dependencies | Upgrade direct dependencies`

### PR DESCRIPTION
**Pull request by mfussenegger**
Before:

    Lens1Lens2

After:

    Lens1 | Lens2

Fixes https://github.com/neovim/neovim/issues/15048

### CODE REVIEW

**Comment by jan-xyz** [comment]
You honestly _did not_ just made a Pull Request 1h after I reported this 😭 ❤️

**Review by mjlbach — APPROVED** [review]
LGTM, also tested it and it addresses the issue.

File: runtime/lua/vim/lsp/codelens.lua [hunk 1]
```diff
@@ -111,12 +111,16 @@ function M.display(lenses, bufnr, client_id)
...
+    local line_lenses = lenses_by_lnum[i] or {}
     api.nvim_buf_clear_namespace(bufnr, ns, i, i + 1)
     local chunks = {}
-    for _, lens in pairs(line_lenses or {}) do
+    local num_line_lenses = #line_lenses
+    for j, lens in ipairs(line_lenses) do
       local text = lens.command and lens.command.title or 'Unresolved lens ...'
       table.insert(chunks, {text, 'LspCodeLens' })
+      if j < num_line_lenses then
+        table.insert(chunks, {' | ', 'LspCodeLens' })
```
Comment (bew) [inline runtime/lua/vim/lsp/codelens.lua:122]: What about a `LspCodeLensSeparator` hi group?

Comment (mfussenegger) [reply runtime/lua/vim/lsp/codelens.lua:122]: Good idea, I'll change it :+1:

**Comment by mjlbach** [comment]
@mfussenegger make sure you squash your fixup commit but then you are good to merge once tests pass.

**Comment by mfussenegger** [comment]
Test failure is unrelated:

```
  [----------] Running tests from test/functional/core/channels_spec.lua
  [ RUN      ] channels can connect to socket: 0.03 ms SKIP
  [ RUN      ] channels can use stdio channel: 38.58 ms OK
  [ RUN      ] channels can use stdio channel with pty: 24.30 ms ERR
  test/helpers.lua:73: Expected objects to be the same.
  Passed in:
  (table: 0x7f13c1c6ba58) {
    [1] = 'notification'
    [2] = 'stdout'
   *[3] = {
      [1] = 3
     *[2] = {
       *[1] = 'i' } } }
  Expected:
  (table: 0x7f13b7baccb8) {
    [1] = 'notification'
    [2] = 'stdout'
   *[3] = {
      [1] = 3
     *[2] = {
       *[1] = 'incomplet' } } }
```

**Pull request merged**